contribute/design/1.x/cstor_cr_schema/apis.md

```diff
@@ -0,0 +1,550 @@
+	BlockDeviceName string `json:"blockDeviceName"`
+	// Capacity is the capacity of the block device.
+	// It is system generated
+	Capacity string `json:"capacity"`
+	// DevLink is the dev link for block devices
+	DevLink string `json:"devLink"`
+}
+
+// CStorPoolClusterStatus represents the latest available observations of a CSPC's current state.
+type CStorPoolClusterStatus struct {
+	// CurrentNumberInstances is the the number of CSPI present at the current state. 
+	CurrentNumberInstances int32 `json:"currentNumberInstances"`
+	
+	// DesiredNumberInstances is the number of CSPI that should be present in the system.
+	DesiredNumberInstances int32 `json:"desiredNumberInstances"`
+	
+	// HealthyNumberInstances is the number of Healthy CSPI present in the system.
+	HealthyNumberInstances int32 `json:"healthyNumberInstances"`
+
+	// Current state of CSPC.
+	Conditions []CStorPoolClusterCondition 
+}
+
+type CSPCConditionType string
+
+// CStorPoolClusterCondition describes the state of a CSPC at a certain point.
+type CStorPoolClusterCondition struct {
+	// Type of CSPC condition.
+	Type CSPCConditionType `json:"type" protobuf:"bytes,1,opt,name=type,casttype=DeploymentConditionType"`
+	// Status of the condition, one of True, False, Unknown.
+	Status corev1.ConditionStatus `json:"status" protobuf:"bytes,2,opt,name=status,casttype=k8s.io/api/core/v1.ConditionStatus"`
+	// The last time this condition was updated.
+	LastUpdateTime metav1.Time `json:"lastUpdateTime,omitempty" protobuf:"bytes,6,opt,name=lastUpdateTime"`
+	// Last time the condition transitioned from one status to another.
+	LastTransitionTime metav1.Time `json:"lastTransitionTime,omitempty" protobuf:"bytes,7,opt,name=lastTransitionTime"`
+	// The reason for the condition's last transition.
+	Reason string `json:"reason,omitempty" protobuf:"bytes,4,opt,name=reason"`
+	// A human readable message indicating details about the transition.
+	Message string `json:"message,omitempty" protobuf:"bytes,5,opt,name=message"`
+}
+
+// CStorPoolClusterList is a list of CStorPoolCluster resources
+type CStorPoolClusterList struct {
+	metav1.TypeMeta `json:",inline"`
+	metav1.ListMeta `json:"metadata"`
+
+	Items []CStorPoolCluster `json:"items"`
+}
+
+```
+
+### CSPI Schema
+
+The CSPI resuses the CSPC schema and following is the current schema in go struct.
+
+```go
+// CStorPoolInstance describes a cstor pool instance resource created as custom resource.
+type CStorPoolInstance struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec           CStorPoolInstanceSpec `json:"spec"`
+	Status         CStorPoolStatus       `json:"status"`
+	VersionDetails VersionDetails        `json:"versionDetails"`
+}
+
+// CStorPoolInstanceSpec is the spec listing fields for a CStorPoolInstance resource.
+type CStorPoolInstanceSpec struct {
+	// HostName is the name of kubernetes node where the pool
+	// should be created.
+	HostName string `json:"hostName"`
```

Comment:
why one is HostName and other is NodeSelector? can we make both of them look same as hostName can be different from NodeName?

Reply:
hostName is the actual hostname that is filled by cspc operator in the cspi which is found via nodeselector labels.
The nodeSelector is just being put into CSPI for informational purpose.